R/bcdc_search.R

```diff
@@ -236,3 +236,24 @@ as.bcdc_recordlist <- function(x) {
   class(x) <- "bcdc_recordlist"
   x
 }
+
+
+#' Provide metadata for all the resources from a single B.C. Data Catalogue record
+#'
+#' @param x a list containing the metadata for a record
```

Comment:
I think elsewhere we just call this a `bcdc_record` object rather than a "list containing..."

Reply:
I think to be consistent, this argument should be called `record`

Reply:
`record` is used where the input is ```either a `bcdc_record` object (from the result of `bcdc_get_record()`),
#' a character string denoting the name or ID of a resource (or the URL) or a BC Geographic
#' Warehouse (BCGW) name.``` Since this extractor function only takes a `bcdc_record` at this time, I propose we call the argument `bcdc_record`?

Reply:
Q: Is the vision for it to eventually take the name or ID? It would be nice to guard against future changes to the argument name.

Reply:
I think it's best to keep the argument name consistent as well (i.e., `record`). The error is informative if the user gives a character (record name/id), and as @boshek says it means it will be consistent in the future.

Also, it might be worth adding the `bcdc_tidy_resources.character()` method now while it's fresh... .  I.e., would need to add 

- a generic (`bcdc_tidy_resources()`), 
- a default method (`bcdc_tidy_resources.default()` that just throws an error), 
- a character method (`bcdc_tidy_resources.character()`) that calls `bcdc_get_record()` and then `bcdc_tidy_resources()` on the returned record, 
- change the current function to a bcdc_record method (`bcdc_tidy_resources.bcdc_record()`)

An example pattern: https://github.com/bcgov/bcdata/blob/1302ea9d3ede1a24cf204ce3ad1468f0f0a26f0c/R/describe-feature.R#L28-L53

Reply:
🤯

Reply:
Sorry! It totally doesn't have to be done now, but I wanted to outline how we could do it whenever one of us has the time

Reply:
I gave it a go!